wallets-testing/pages/iframeWidget.page.ts

```diff
@@ -0,0 +1,175 @@
+import { expect, FrameLocator, Locator, Page, test } from '@playwright/test';
+import { ConsoleLogger } from '@nestjs/common';
+import {
+  WalletPage,
+  WalletConnectTypes,
+} from '@lidofinance/wallets-testing-wallets';
+import { BrowserService } from '@lidofinance/browser-service';
+import { WidgetPage } from './widget.page';
+import { WidgetConfig } from '../config';
+
+export class IframeWidgetPage implements WidgetPage {
+  logger = new ConsoleLogger(IframeWidgetPage.name);
+  walletPage: WalletPage<WalletConnectTypes.IFRAME>;
+  app: FrameLocator;
+  page: Page;
+
+  wrapTabBtn: Locator;
+
+  connectBtn: Locator;
+  stakeInput: Locator;
+  stakeSubmitBtn: Locator;
+  enabledStakeSubmitBtn: Locator;
+
+  headerAccountSection: Locator;
+  providerName: Locator;
+  ethAvailableToStakeValue: Locator;
+  termsCheckbox: Locator;
+  copyWcUrlBtn: Locator;
+  closeAccountModalBtn: Locator;
+
+  wrapInput: Locator;
+  wrapSubmitBtn: Locator;
+  enabledWrapSubmitBtn: Locator;
+
+  constructor(
+    browserService: BrowserService,
+    public widgetConfig: WidgetConfig,
+  ) {
+    this.walletPage = browserService.getWalletPage();
+    this.page = browserService.getBrowserContextPage();
+    this.app = this.page.locator('iframe').first().contentFrame();
+
+    this.wrapTabBtn = this.app.getByTestId('navWrap');
+
+    this.connectBtn = this.app
+      .getByRole('button')
+      .getByText('Connect', { exact: true });
+    this.stakeInput = this.app.getByTestId('stakeInput');
+    this.stakeSubmitBtn = this.app.getByTestId('stakeSubmitBtn');
+    this.enabledStakeSubmitBtn = this.app.locator(
+      'button[data-testid="stakeSubmitBtn"]:not([disabled])',
+    );
+
+    this.headerAccountSection = this.app.getByTestId('accountSectionHeader');
+    this.providerName = this.app.locator('div[data-testid="providerName"]');
+    this.ethAvailableToStakeValue = this.app.getByTestId('ethAvailableToStake');
+    this.termsCheckbox = this.app.getByRole('checkbox');
+    this.copyWcUrlBtn = this.app.locator('.wcm-action-btn');
+    this.closeAccountModalBtn = this.app
+      .locator('div[role="dialog"] button')
+      .nth(0);
+
+    this.wrapInput = this.app.getByTestId('wrapInput');
+    this.wrapSubmitBtn = this.app.getByTestId('wrapBtn');
+    this.enabledWrapSubmitBtn = this.app.locator(
+      'button[data-testid="wrapBtn"]:not([disabled])',
+    );
+  }
+
+  async connectWallet() {
+    return await test.step('Connect wallet to Lido app in Safe', async () => {
+      const attemptsToConnect = 3;
+      for (let attempt = 1; attempt <= attemptsToConnect; attempt++) {
```

Comment:
Could you add comment why do we need attempts here to remove it once iSafe become more stable for autotests?